crates/cargo-wdk/src/actions/build/package_task.rs

```diff
@@ -631,4 +650,77 @@ mod tests {
 
         PackageTask::new(package_task_params, &wdk_build, &command_exec, &fs);
     }
+
+    #[test]
+    fn stampinf_version_overrides_with_env_var() {
+        use std::process::{ExitStatus, Output};
+
+        // verify both with and without the env var set scenarios
+        let scenarios = [("env_set", Some("1.2.3.4")), ("env_unset", None)];
+
+        // Compile-time evaluation: cfg!(...) => true only if
+        // `allow_stampinf_version_env_override` was enabled during build
+        let is_cfg_set = cfg!(allow_stampinf_version_env_override);
+
+        for (name, env_val) in scenarios {
+            if let Some(v) = env_val {
+                std::env::set_var("STAMPINF_VERSION", v);
```

Comment:
may race with changes in #430 depending on who gets their changes in first. fyi @leon-xd